Lib/test/_test_multiprocessing.py

```diff
@@ -5760,6 +5760,33 @@ def test_namespace(self):
         self.run_worker(self._test_namespace, o)
 
 
+class TestNamedResource(unittest.TestCase):
+    def test_global_named_resource_spawn(self):
+        #
+        # Check that global named resources in main module
+        # will not leak by a subprocess, in spawn context.
+        #
+        with os_helper.temp_dir() as tmp_dir:
+            source = os.path.join(tmp_dir, 'source.py')
+            with open(source, 'w') as f:
+                f.write('''if 1:
```

Comment:
You can use textwrap.dedent() to avoid ``if 1:``.

Reply:
I've changed it to use textwrap.dedent().